### ISSUE TITLE
nullcontext: untyped or Any enter_result type is ignored and inferred as None on __enter__

### ISSUE THREAD

**Issue by RomanValov** [microsoft/pyright#11304]
**Describe the bug**
The `nullcontext` context manager from stdlib lets to pass arbitrary value to be returned on `__enter__`. It's named `enter_result`.
If the value passed is untyped or `Any`, pyright infers return type of `__enter__` as `None`. But should preserve untyped/`Any` type.

**Code or Screenshots**
```python
from contextlib import nullcontext
from typing import Any, revela_type

# also with: def test(items) -> None:
def test(items: Any) -> None:
    with nullcontext(items) as _items:
        reveal_type(_items)
        for item in _items:
            print(item)


if __name__ == '__main__':
    test([1, 2, 3])
```

**Expected behavior**
Revealed type is Any, no errors

**Observed behavior**
```sh
$ pyright ./test.py
test.py
  test.py:9:21 - information: Type of "_items" is "None"
  test.py:10:21 - error: "None" is not iterable
    "__iter__" method not defined (reportGeneralTypeIssues)
1 error, 0 warnings, 1 information
```

**VS Code extension or command-line**
CLI pyright 1.1.408
